docs/DeveloperGuide.md

```diff
@@ -4,26 +4,231 @@
+
+`find` has the following options:
+- `find /id [ID]`
+- `find /t [TITLE]`
+- `find /a [AUTHOR]`
+- `find /i [ISBN]`
+
+Multiple filters can also be combined:
+
+- `find /t [TITLE] /a [AUTHOR]`
+
+#### Implementation
+
+Upon receiving the `find` command, the system will:
+
+1. Parse the filters and their associated values.
+2. Filter the resources based on the given filters.
+3. Display the matching resources.
+
+#### Example Usage Scenario
+
+**Step 1.** The user inputs the command: `find /a "F. Scott Fitzgerald"`
+
+**Step 2.** The `UI` component forwards the input to `SYSLIB`, which then sends it to the `PARSER`.
+
+**Step 3.** The `PARSER` processes the command, extracts the `author` filter, and retrieves all resources written by "F. Scott Fitzgerald".
+
+**Step 4.** The matching resources are displayed to the user.
+
+#### Sequence Diagram
+
+<img src="images/FindSequenceDiagram.png" />
+
+### Examples for Testing
+
+1. **Find by Author**
+    - Test case: `find /a "F. Scott Fitzgerald"`
+
+      Expected: All resources written by F. Scott Fitzgerald are displayed.
+
+2. **Find by ISBN**
+    - Test case: `find /i "9780061120084"`
+
+      Expected: The resource with ISBN "9780061120084" is displayed, which should be "To Kill a Mockingbird" by Harper Lee.
+
+3. **Find by ID**
+    - Test case: `find /id 2`
+
+      Expected: The resource with ID "2" is displayed, which should be "To Kill a Mockingbird" by Harper Lee.
+
+4. **Find by Title**
+    - Test case: `find /t "The Great Gatsby"`
+
+      Expected: The resource titled "The Great Gatsby" is displayed.
+
+5. **Combining Filters**
+    - Test case: `find /a "F. Scott Fitzgerald" /t "The Great Gatsby"`
+
+      Expected: Resources that match either the title "The Great Gatsby" or the author "F. Scott Fitzgerald" are displayed.
+
+
+### Add Resource Feature
+
+The `add` feature is responsible for processing user commands to add a new book to SysLib. It is facilitated by 
+the `AddCommand` component. It works with `Parser` and `Command` components to parse and validate the user input. 
+The new book is stored internally in `resourceList` as a `Book`. 
+
+`add` has seven options:
+- add /id [id] /t [title] /a [author] /tag [tag] /i [isbn]
+- add /id [id] /t [title] /a [author] /tag [tag] /i [isbn] _/g [genre] /s [status]_
+
+#### Implementation
+
+It implements the following operations:
+
+- `ADDCOMMAND#parseArgument(statement: String)` -- Parses the input command to extract relevant information.
+- `ADDCOMMAND#validate(statement: String, values: String[])` -- Validates the input statement to ensure that it is valid.
+- `ADDCOMMAND#createBook(values: String[])` -- Creates a new book based on the parsed and validated values.
+
+#### Example Usage Scenario
+
+Step 1. The user inputs the command: `add /id 0005 /t Frankenstein /a Mary Shelley /i FKS0001 /tag B /g Gothic, Fiction`
```

Comment:
Why is "Step 1." in bold above but not here? Is it better to have a unified style?